### DISCUSSION TITLE
What in the world is a shim?

### DISCUSSION THREAD

**Discussion by gsal** [ScoopInstaller/Scoop#5205] [General]
Hi, just installed scoop, last night....still reading, browsing contents, setup, and familiarizing myself with it. 

So, I have scoop

- Installed `make`
- When I type `make` at the prompt, I presume `make.exe` is executed? 
- Then what? 
    - Is `make.shim` picked up? 
    - read? 
    - interpreted? 
    - what's the syntax? language? 
    - what kind of things can and cannot do in there?  

In particular, I would like to install scoop in a LAN drive which can then be mapped by the users; but, it could be mapped under different drive letters. Then, I am thinking I could ask users to set an environment variable indicating which letter they will be using and then, inside the shim, I would like to see something like: 
```
path = "%SCOOP_HOME%\scoop\apps\make\current\bin\make.exe"
```
I tested setting `SCOOP_HOME` to its current installation spot in my home directory, but it does not like the environment variable thing. 

What's the correct way to go about it?

**Comment by gsal** [comment]
Well, just to report that after digging a bit, I now know what a shim is  ;- )

- looked into command `scoop shim`
- dug down a bit 
- found a `Makefile` with reference to github project `scoop-better-shimexe`
- downloaded the project

**Issue**:
All shim files have the path to `<scoop_home>\apps` hard-coded into them
```
$ cd scoop/shims
$ cat *.shim
path = "C:\Users\<userid>\scoop\apps\7zip\current\7z.exe"
path = "C:\Users\<userid>\scoop\apps\7zip\current\7zFM.exe"
path = "C:\Users\<userid>\scoop\apps\7zip\current\7zG.exe"
.
.
.
```

This seems redundant and unnecessary; and, it keeps the scoop installation from being portable.

**Proposal**:
- Let `path` in *.shim files **_not_** have to include the common portion all the way to `apps` and only specify what is unique, i.e., 
`path = "7zip\current\7z.exe"`
`path = "7zip\current\7zFM.exe"`
`path = "7zip\current\7zG.exe"`
- Modigy shim.c, so that it first determines its own path, presumably `<scoop_home>\shims\<app.exe>`
- Remove the last portion starting with \shims so that scoop_gome=<scoop_home>
- append apps\
- append remaining <path> found in the *.shim file
- finally: `cmd = <scoop_home> + \ + apps + \ + <path>`

With my just-about-zero experience with scoop, I tested this and works; but, I have no idea it this breaks and/or get in the way of aspects of scoop.

Anyway, just an idea to remove at least one aspect that may be keeping scoop from being portable.

**Comment by bernardoadc** [comment]
Is this thread dead? Why no official comments on it?

**Reply by HUMORCE** [reply]
Which part? 

For what is shim: https://stackoverflow.com/questions/2116142/what-is-a-shim

For re-fine shim after migration: `scoop reset *`

For network shared drive: I have never installed Scoop on network shared drive, but according to many related threads, Scoop doesn't work properly on the network share drive and non-NTFS drive.

**Reply by hrxn** [reply]
If the network drive is mounted/mapped properly, it should definitely work.
